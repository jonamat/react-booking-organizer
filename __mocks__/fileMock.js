// Mocks static assets imported from file-loader ex. jpe?g, png, svg etc
module.exports = 'test-file-stub';
