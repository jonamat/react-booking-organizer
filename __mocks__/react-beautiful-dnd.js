module.exports = {
    Droppable: ({ children }) =>
        children(
            {
                draggableProps: {
                    style: {},
                },
                innerRef: () => undefined,
            },
            {},
        ),
    Draggable: ({ children }) =>
        children(
            {
                draggableProps: {
                    style: {},
                },
                innerRef: () => undefined,
            },
            {},
        ),
    DragDropContext: ({ children }) => children,
};
