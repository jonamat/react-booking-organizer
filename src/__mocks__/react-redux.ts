import { Customer, Employee, Reservation, Service } from '../types';

export const __mockCustomers: Array<Customer> = [
    {
        id: 'customer-FAKE-ID1',
        name: 'Customer name',
        surname: 'Surname',
    },
    {
        id: 'customer-FAKE-ID2',
        name: 'Other name',
        surname: 'Other surname',
    },
];

export const __mockEmployees: Array<Employee> = [
    {
        id: 'employee-FAKE-ID1',
        name: 'Employee name 1',
    },
    {
        id: 'employee-FAKE-ID2',
        name: 'Employee name 2',
    },
];

export const __mockServices: Array<Service> = [
    {
        id: 'service-FAKE-ID1',
        name: 'fake service 1',
        averageDuration: 30,
    },
    {
        id: 'service-FAKE-ID2',
        name: 'fake service 2',
        averageDuration: 45,
    },
];

export const __mockReservations: Array<Reservation> = [
    {
        id: 'reservation-FAKE-ID1',
        date: new Date('1 Jan 2000 10:00 GMT+0100'),
        serviceId: __mockServices[0].id,
        customerId: __mockCustomers[0].id,
        employeeId: __mockEmployees[0].id,
        notes: 'something',
    },
    {
        id: 'reservation-FAKE-ID2',
        date: new Date('1 Jan 2040 10:00 GMT+0100'),
        serviceId: __mockServices[1].id,
        customerId: __mockCustomers[0].id,
        employeeId: __mockEmployees[1].id,
        notes: 'something2',
    },
];

export let useStore = () => ({
    getState: () => ({
        database: {
            customers: __mockCustomers,
            employees: __mockEmployees,
            services: __mockServices,
            reservations: __mockReservations,
        },
    }),
});

export const __setUseStore = (value) => (useStore = value);
export const useDispatch = () => undefined;
export const connect = () => (component) => component;
