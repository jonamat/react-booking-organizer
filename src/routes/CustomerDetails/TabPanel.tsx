import React, { FC } from 'react';
import { Typography } from '@material-ui/core';

interface TabPanelProps {
    children?: React.ReactNode;
    index: any;
    value: any;
}

const TabPanel: FC<TabPanelProps> = ({ children, value, index, ...args }) => {
    return (
        <Typography component="div" role="tabpanel" hidden={value !== index} {...args}>
            {value === index && children}
        </Typography>
    );
};

export default TabPanel;
