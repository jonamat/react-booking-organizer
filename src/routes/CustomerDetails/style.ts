import { makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) => ({
    container: {
        padding: theme.spacing(2),
    },
    header: {
        display: 'flex',
        paddingTop: theme.spacing(2),
        paddingBottom: theme.spacing(2),
    },
    grow: {
        flexGrow: 1,
    },
}));

export default useStyles;
