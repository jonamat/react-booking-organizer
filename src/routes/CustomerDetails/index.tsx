import React, { FC, useState } from 'react';
import { RouteComponentProps } from 'react-router';
import { bindActionCreators } from 'redux';
import { useStore, useDispatch } from 'react-redux';
import { toast } from 'react-toastify';
import {
    Typography,
    IconButton,
    Tabs,
    Tab,
    Grid,
    List,
    ListItem,
    ListItemIcon,
    ListItemText,
    DialogContent,
    DialogContentText,
    DialogActions,
    Button,
    CircularProgress,
    Tooltip,
} from '@material-ui/core';

import { openDialog, closeDialog } from '../../redux';
import { Reservation, RootState } from '../../types';
import { deleteCustomer } from '../../utils';
import ReservationPanel from '../../components/ReservationPanel';
import Center from '../../components/Center';
import TabPanel from './TabPanel';
import useStyles from './style';

// Icons
import DeleteRoundedIcon from '@material-ui/icons/DeleteRounded';
import EditRoundedIcon from '@material-ui/icons/EditRounded';
import AddCircleOutlineRoundedIcon from '@material-ui/icons/AddCircleOutlineRounded';
import AssignmentIndRoundedIcon from '@material-ui/icons/AssignmentIndRounded';
import PersonPinCircleRoundedIcon from '@material-ui/icons/PersonPinCircleRounded';
import PhoneRoundedIcon from '@material-ui/icons/PhoneRounded';
import PostAddRoundedIcon from '@material-ui/icons/PostAddRounded';
import AssignmentTurnedInRoundedIcon from '@material-ui/icons/AssignmentTurnedInRounded';
import DateRangeRoundedIcon from '@material-ui/icons/DateRangeRounded';

const CustomerDetails: FC<RouteComponentProps<{ customerId: string }>> = ({ match, history }) => {
    const classes = useStyles();
    const now = Date.now();
    const [tabIndex, setTabIndex] = useState<number>(0);
    const {
        database: { reservations, services, customers },
    } = useStore<RootState>().getState();
    const actions = bindActionCreators({ openDialog, closeDialog }, useDispatch());

    if (!reservations || !services || !customers) throw new Error('Internal error - Resources loading logic broken');

    const customerId = match.params.customerId;
    const customer = customers.find((customer) => customer.id === customerId);

    if (!customer) {
        history.push('/customers');
        toast.error('Cliente non presente del database');
        return null;
    }

    // All the reservations assigned to the customer
    const personalReservation = reservations.filter((reservation) => reservation.customerId === customer.id);

    // All the customer reservation from NOW
    const futureReservations: Array<Reservation> = [];

    // All the customer reservation until NOW
    const pastReservations: Array<Reservation> = personalReservation
        .reduce<Array<Reservation>>((prev, curr) => {
            if (curr.date.getTime() < now) prev.push(curr);
            else futureReservations.push(curr);
            return prev;
        }, [])
        .sort((a, b) => b.date.getTime() - a.date.getTime());
    futureReservations.sort((a, b) => a.date.getTime() - b.date.getTime());

    const handleDeleteCustomer = () => {
        actions.openDialog({
            title: 'Eliminazione in corso',
            content: (
                <Center style={{ height: 100 }}>
                    <CircularProgress />
                </Center>
            ),
        });

        deleteCustomer(reservations, customer)
            .then(() => {
                history.push('/customers');
                toast.success('Cliente eliminato');
            })
            .catch((error) => toast.error(error.message))
            .finally(() => actions.closeDialog());
    };

    const confirmationDialog = (
        <>
            <DialogContent>
                <DialogContentText>
                    Eliminando il cliente eliminerai anche tutti gli appuntamenti passati e futuri assegnati ad esso.
                    Non sar√† possibile recuperare tali informazioni.
                </DialogContentText>
            </DialogContent>
            <DialogActions>
                <Button onClick={() => actions.closeDialog()} color="primary" autoFocus>
                    Annulla
                </Button>
                <Button onClick={handleDeleteCustomer} color="primary" data-testid="delete-confirmation">
                    Elimina
                </Button>
            </DialogActions>
        </>
    );

    return (
        <div className={classes.container}>
            <Grid container justify="space-between" alignItems="center" wrap="nowrap">
                <Grid item>
                    <Typography variant="h6">Scheda cliente</Typography>
                </Grid>

                <div className={classes.grow} />
                <Grid item>
                    <Tooltip title="Assegna nuovo appuntamento">
                        <IconButton
                            onClick={() => history.push('/add-reservation?customerId=' + customer.id)}
                            data-testid="add-reservation"
                        >
                            <AddCircleOutlineRoundedIcon color="primary" />
                        </IconButton>
                    </Tooltip>
                </Grid>

                <Grid item>
                    <Tooltip title="Modifica">
                        <IconButton
                            onClick={() => history.push('/edit-customer/' + customer.id)}
                            data-testid="edit-customer"
                        >
                            <EditRoundedIcon color="primary" />
                        </IconButton>
                    </Tooltip>
                </Grid>

                <Grid item>
                    <Tooltip title="Elimina cliente">
                        <IconButton
                            onClick={() =>
                                actions.openDialog({ title: 'Elimina cliente', content: confirmationDialog })
                            }
                            data-testid="delete-customer"
                        >
                            <DeleteRoundedIcon color="primary" />
                        </IconButton>
                    </Tooltip>
                </Grid>
            </Grid>

            <Grid container>
                <Grid item xs={12} sm={6}>
                    <List disablePadding>
                        {[
                            {
                                label: 'Nome',
                                value: customer.name + ' ' + customer.surname || '',
                                icon: <AssignmentIndRoundedIcon color="primary" />,
                            },
                            {
                                label: 'Identificatore aggiuntivo',
                                value: customer.additionalIdentifier,
                                icon: <PersonPinCircleRoundedIcon color="primary" />,
                            },
                            {
                                label: 'Telefono',
                                value: customer.phone,
                                icon: <PhoneRoundedIcon color="primary" />,
                            },
                        ].map((item, index) => (
                            <ListItem key={index}>
                                <ListItemIcon>{item.icon}</ListItemIcon>
                                <ListItemText primary={item.value || '---'} secondary={item.label} />
                            </ListItem>
                        ))}
                    </List>
                </Grid>

                <Grid item xs={12} sm={6}>
                    <List disablePadding>
                        {[
                            {
                                label: 'Note',
                                value: customer.notes,
                                icon: <PostAddRoundedIcon color="primary" />,
                            },
                            {
                                label: 'Appuntamenti passati',
                                value: pastReservations.length,
                                icon: <AssignmentTurnedInRoundedIcon color="primary" />,
                            },
                            {
                                label: 'Appuntamenti futuri',
                                value: futureReservations.length,
                                icon: <DateRangeRoundedIcon color="primary" />,
                            },
                        ].map((item, index) => (
                            <ListItem key={index}>
                                <ListItemIcon>{item.icon}</ListItemIcon>
                                <ListItemText primary={item.value || '---'} secondary={item.label} />
                            </ListItem>
                        ))}
                    </List>
                </Grid>
            </Grid>

            <Tabs variant="fullWidth" centered value={tabIndex} onChange={(_e, index) => setTabIndex(index)}>
                <Tab label="Appuntamenti passati" />
                <Tab label="Appuntamenti futuri" />
            </Tabs>
            <TabPanel value={tabIndex} index={0} data-testid="past-reservations">
                <Grid container spacing={1}>
                    {pastReservations.length ? (
                        pastReservations.map((reservation) => (
                            <Grid key={reservation.id} item xs={12}>
                                <ReservationPanel reservation={reservation} showDetails />
                            </Grid>
                        ))
                    ) : (
                        <Grid item>
                            <Typography variant="body1">Nessun appuntamento presente</Typography>
                        </Grid>
                    )}
                </Grid>
            </TabPanel>
            <TabPanel value={tabIndex} index={1}>
                <Grid container spacing={1}>
                    {futureReservations.length ? (
                        futureReservations.map((reservation) => (
                            <Grid key={reservation.id} item xs={12}>
                                <ReservationPanel reservation={reservation} showDetails />
                            </Grid>
                        ))
                    ) : (
                        <Grid item>
                            <Typography variant="body1">Nessun appuntamento presente</Typography>
                        </Grid>
                    )}
                </Grid>
            </TabPanel>
        </div>
    );
};

export default React.memo(
    CustomerDetails,
    (prevProps, nextProps) => prevProps.location.pathname === nextProps.location.pathname,
);
