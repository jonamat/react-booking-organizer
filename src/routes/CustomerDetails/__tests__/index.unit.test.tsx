/**
 * Unit test suite
 *
 * Target: CustomerDetails component
 * Tests:
 * - Data position correctness
 * - Redirection
 * - Snapshot matching
 * - Functionality
 */

// Test resources
import React from 'react';
import renderer from 'react-test-renderer';
import { StylesProvider, createGenerateClassName } from '@material-ui/core';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// Test configuration
import { __mockCustomers } from '../../../__mocks__/react-redux';

// Component to test
import CustomerDetails from '..';

const match = {
    params: {
        customerId: __mockCustomers[0].id,
    },
};

/* -------------------------------------------------------------------------- */
/*                                    Mocks                                   */
/* -------------------------------------------------------------------------- */

jest.mock('../../../utils', () => ({
    deleteCustomer: jest.fn(() => Promise.resolve()),
}));
jest.mock('../../../components/ReservationPanel', () => () => null);
const mockOpenDialog = jest.fn();
jest.mock('redux', () => ({
    ...(jest.requireActual('redux') as Record<string, any>),
    bindActionCreators: () => ({
        openDialog: mockOpenDialog,
        closeDialog: () => undefined,
    }),
}));
const history = {
    push: jest.fn(),
};

/* -------------------------------------------------------------------------- */
/*                              Bridge operations                             */
/* -------------------------------------------------------------------------- */

beforeEach(() => {
    jest.clearAllMocks();
});

/* -------------------------------------------------------------------------- */
/*                               Render wrapper                               */
/* -------------------------------------------------------------------------- */

/**
 * Render the component with the custom configuration
 * @param props Component props
 */
const renderComponent = (props?) => {
    return render(<CustomerDetails history={history} match={match} {...props} />);
};

/* -------------------------------------------------------------------------- */
/*                                    Tests                                   */
/* -------------------------------------------------------------------------- */

describe('CustomerDetails', () => {
    describe('Data accuracy', () => {
        it('shows correct data', () => {
            const { getByText, queryAllByText } = renderComponent();

            expect(getByText('Nome').previousElementSibling).toHaveTextContent(
                __mockCustomers[0].name + ' ' + __mockCustomers[0].surname || '',
            );
            expect(getByText('Identificatore aggiuntivo').previousElementSibling).toHaveTextContent(
                __mockCustomers[0].additionalIdentifier || '---',
            );
            expect(getByText('Telefono').previousElementSibling).toHaveTextContent(__mockCustomers[0].phone || '---');
            expect(getByText('Note').previousElementSibling).toHaveTextContent(__mockCustomers[0].notes || '---');
            expect(queryAllByText('Appuntamenti passati')[0].previousElementSibling).toHaveTextContent('1');
            expect(queryAllByText('Appuntamenti futuri')[0].previousElementSibling).toHaveTextContent('1');
        });
    });

    describe('User experience', () => {
        it('redirects users at "Add Reservation" button click', () => {
            const { getByTestId } = renderComponent();
            fireEvent.click(getByTestId('add-reservation'));
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith('/add-reservation?customerId=' + __mockCustomers[0].id);
        });

        it('redirects users at "Edit" button click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('edit-customer'));
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith('/edit-customer/' + __mockCustomers[0].id);
        });

        it('display confirmation dialog at "Delete" button click', () => {
            const { getByTestId } = renderComponent();

            fireEvent.click(getByTestId('delete-customer'));

            expect(mockOpenDialog).toBeCalledTimes(1);
        });
    });

    it('matches snapshot', () => {
        const snapshot = renderer
            .create(
                <StylesProvider generateClassName={createGenerateClassName({ disableGlobal: true })}>
                    {/* @ts-ignore */}
                    <CustomerDetails match={match} />
                </StylesProvider>,
            )
            .toJSON();
        expect(snapshot).toMatchSnapshot();
    });
});
