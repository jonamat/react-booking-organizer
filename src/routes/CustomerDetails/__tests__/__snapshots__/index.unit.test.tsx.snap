// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerDetails matches snapshot 1`] = `
<div
  className="jss1"
>
  <div
    className="jss4 jss5 jss11 jss13 jss24"
  >
    <div
      className="jss4 jss6"
    >
      <h6
        className="jss107 jss117"
      >
        Scheda cliente
      </h6>
    </div>
    <div
      className="jss3"
    />
    <div
      className="jss4 jss6"
    >
      <button
        aria-describedby={null}
        className="jss157 jss148"
        data-testid="add-reservation"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Assegna nuovo appuntamento"
        type="button"
      >
        <span
          className="jss156"
        >
          <svg
            aria-hidden={true}
            className="jss160 jss161"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c-.55 0-1 .45-1 1v3H8c-.55 0-1 .45-1 1s.45 1 1 1h3v3c0 .55.45 1 1 1s1-.45 1-1v-3h3c.55 0 1-.45 1-1s-.45-1-1-1h-3V8c0-.55-.45-1-1-1zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </span>
        <span
          className="jss217"
        />
      </button>
    </div>
    <div
      className="jss4 jss6"
    >
      <button
        aria-describedby={null}
        className="jss157 jss148"
        data-testid="edit-customer"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Modifica"
        type="button"
      >
        <span
          className="jss156"
        >
          <svg
            aria-hidden={true}
            className="jss160 jss161"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.46v3.04c0 .28.22.5.5.5h3.04c.13 0 .26-.05.35-.15L17.81 9.94l-3.75-3.75L3.15 17.1c-.1.1-.15.22-.15.36zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          className="jss217"
        />
      </button>
    </div>
    <div
      className="jss4 jss6"
    >
      <button
        aria-describedby={null}
        className="jss157 jss148"
        data-testid="delete-customer"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Elimina cliente"
        type="button"
      >
        <span
          className="jss156"
        >
          <svg
            aria-hidden={true}
            className="jss160 jss161"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v10zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
            />
          </svg>
        </span>
        <span
          className="jss217"
        />
      </button>
    </div>
  </div>
  <div
    className="jss4 jss5"
  >
    <div
      className="jss4 jss6 jss50 jss58"
    >
      <ul
        className="jss169"
      >
        <li
          className="jss173 jss180"
          disabled={false}
        >
          <div
            className="jss184"
          >
            <svg
              aria-hidden={true}
              className="jss160 jss161"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm0 4c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm6 12H6v-1.4c0-2 4-3.1 6-3.1s6 1.1 6 3.1V19z"
              />
            </svg>
          </div>
          <div
            className="jss186 jss187"
          >
            <span
              className="jss107 jss190 jss109 jss136"
            >
              Customer name Surname
            </span>
            <p
              className="jss107 jss191 jss108 jss133 jss136"
            >
              Nome
            </p>
          </div>
        </li>
        <li
          className="jss173 jss180"
          disabled={false}
        >
          <div
            className="jss184"
          >
            <svg
              aria-hidden={true}
              className="jss160 jss161"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 11c-1.33 0-4 .67-4 2v.16c.97 1.12 2.4 1.84 4 1.84s3.03-.72 4-1.84V13c0-1.33-2.67-2-4-2zm0-1c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0-8c4.2 0 8 3.22 8 8.2 0 3.18-2.45 6.92-7.34 11.23-.38.33-.95.33-1.33 0C6.45 17.12 4 13.38 4 10.2 4 5.22 7.8 2 12 2z"
              />
            </svg>
          </div>
          <div
            className="jss186 jss187"
          >
            <span
              className="jss107 jss190 jss109 jss136"
            >
              ---
            </span>
            <p
              className="jss107 jss191 jss108 jss133 jss136"
            >
              Identificatore aggiuntivo
            </p>
          </div>
        </li>
        <li
          className="jss173 jss180"
          disabled={false}
        >
          <div
            className="jss184"
          >
            <svg
              aria-hidden={true}
              className="jss160 jss161"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.23 15.26l-2.54-.29c-.61-.07-1.21.14-1.64.57l-1.84 1.84c-2.83-1.44-5.15-3.75-6.59-6.59l1.85-1.85c.43-.43.64-1.03.57-1.64l-.29-2.52c-.12-1.01-.97-1.77-1.99-1.77H5.03c-1.13 0-2.07.94-2 2.07.53 8.54 7.36 15.36 15.89 15.89 1.13.07 2.07-.87 2.07-2v-1.73c.01-1.01-.75-1.86-1.76-1.98z"
              />
            </svg>
          </div>
          <div
            className="jss186 jss187"
          >
            <span
              className="jss107 jss190 jss109 jss136"
            >
              ---
            </span>
            <p
              className="jss107 jss191 jss108 jss133 jss136"
            >
              Telefono
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="jss4 jss6 jss50 jss58"
    >
      <ul
        className="jss169"
      >
        <li
          className="jss173 jss180"
          disabled={false}
        >
          <div
            className="jss184"
          >
            <svg
              aria-hidden={true}
              className="jss160 jss161"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 12c-.55 0-1 .45-1 1v5.22c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1h5c.55 0 1-.45 1-1s-.45-1-1-1H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-6c0-.55-.45-1-1-1zM21.02 5H19V2.98c0-.54-.44-.98-.98-.98h-.03c-.55 0-.99.44-.99.98V5h-2.01c-.54 0-.98.44-.99.98v.03c0 .55.44.99.99.99H17v2.01c0 .54.44.99.99.98h.03c.54 0 .98-.44.98-.98V7h2.02c.54 0 .98-.44.98-.98v-.04c0-.54-.44-.98-.98-.98z"
              />
              <path
                d="M14 9H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zM14 12H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1zM14 15H8c-.55 0-1 .45-1 1s.45 1 1 1h6c.55 0 1-.45 1-1s-.45-1-1-1z"
              />
            </svg>
          </div>
          <div
            className="jss186 jss187"
          >
            <span
              className="jss107 jss190 jss109 jss136"
            >
              ---
            </span>
            <p
              className="jss107 jss191 jss108 jss133 jss136"
            >
              Note
            </p>
          </div>
        </li>
        <li
          className="jss173 jss180"
          disabled={false}
        >
          <div
            className="jss184"
          >
            <svg
              aria-hidden={true}
              className="jss160 jss161"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1s-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zM9.29 16.29L6.7 13.7a.9959.9959 0 010-1.41c.39-.39 1.02-.39 1.41 0L10 14.17l5.88-5.88c.39-.39 1.02-.39 1.41 0 .39.39.39 1.02 0 1.41l-6.59 6.59c-.38.39-1.02.39-1.41 0z"
              />
            </svg>
          </div>
          <div
            className="jss186 jss187"
          >
            <span
              className="jss107 jss190 jss109 jss136"
            >
              1
            </span>
            <p
              className="jss107 jss191 jss108 jss133 jss136"
            >
              Appuntamenti passati
            </p>
          </div>
        </li>
        <li
          className="jss173 jss180"
          disabled={false}
        >
          <div
            className="jss184"
          >
            <svg
              aria-hidden={true}
              className="jss160 jss161"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 4h-1V3c0-.55-.45-1-1-1s-1 .45-1 1v1H8V3c0-.55-.45-1-1-1s-1 .45-1 1v1H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 15c0 .55-.45 1-1 1H6c-.55 0-1-.45-1-1V9h14v10zM7 11h2v2H7zm4 0h2v2h-2zm4 0h2v2h-2z"
              />
            </svg>
          </div>
          <div
            className="jss186 jss187"
          >
            <span
              className="jss107 jss190 jss109 jss136"
            >
              1
            </span>
            <p
              className="jss107 jss191 jss108 jss133 jss136"
            >
              Appuntamenti futuri
            </p>
          </div>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="jss192"
  >
    <div
      className="jss197 jss198"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        className="jss194 jss196"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-selected={true}
          className="jss157 jss203 jss205 jss208 jss210"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="jss212"
          >
            Appuntamenti passati
          </span>
          <span
            className="jss217"
          />
        </button>
        <button
          aria-selected={false}
          className="jss157 jss203 jss205 jss210"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="jss212"
          >
            Appuntamenti futuri
          </span>
          <span
            className="jss217"
          />
        </button>
      </div>
      <span
        className="jss213 jss215 jss202"
        style={
          Object {
            "left": 0,
            "width": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="jss107 jss109"
    data-testid="past-reservations"
    hidden={false}
    role="tabpanel"
  >
    <div
      className="jss4 jss5 jss27"
    >
      <div
        className="jss4 jss6 jss50"
      />
    </div>
  </div>
  <div
    className="jss107 jss109"
    hidden={true}
    role="tabpanel"
  />
</div>
`;
